### PR TITLE
Unused bindings part1

### PR DESCRIPTION
**Pull request by dpsutton**
Cleaning up the warnings from `clj-kondo --lint src:shared/src`

Currently

```
clj-kondo --lint src:shared/src | grep -v deprecated | wc -l
      63
```

<details>
<summary> List of non-deprecated warnings</summary>

There are some weird things going on with the `api/defendpoint` macro which is causing the unused binding `p1_23217#` and the other named unused var warnings. Future work with borkdude.
```
❯ clj-kondo --lint src:shared/src | grep -v deprecated
src/metabase/api/database.clj:715:1: warning: unused binding p1__23217#
src/metabase/api/dataset.clj:109:1: warning: unused binding p1__21335#
src/metabase/api/email.clj:68:1: warning: unused binding p1__19748#
src/metabase/api/field.clj:161:1: warning: unused binding field
src/metabase/api/field.clj:184:1: warning: unused binding field
src/metabase/api/metric.clj:79:32: warning: unused binding archived
src/metabase/api/notify.clj:14:1: warning: unused binding table
src/metabase/api/notify.clj:14:1: warning: unused binding database
src/metabase/api/notify.clj:14:1: warning: unused binding db-sync-fn
src/metabase/api/notify.clj:14:1: warning: unused binding table-sync-fn
src/metabase/api/query_description.clj:14:13: warning: unused binding query
src/metabase/api/query_description.clj:66:4: warning: unused binding metadata
src/metabase/api/query_description.clj:71:4: warning: unused binding metadata
src/metabase/api/query_description.clj:88:4: warning: unused binding metadata
src/metabase/api/query_description.clj:89:14: warning: unused binding order-by
src/metabase/api/query_description.clj:96:4: warning: unused binding metadata
src/metabase/api/routes.clj:60:46: warning: Unresolved var: activity/routes
src/metabase/api/routes.clj:61:46: warning: Unresolved var: alert/routes
src/metabase/api/routes.clj:62:46: warning: Unresolved var: magic/routes
src/metabase/api/routes.clj:63:46: warning: Unresolved var: card/routes
src/metabase/api/routes.clj:64:46: warning: Unresolved var: collection/routes
src/metabase/api/routes.clj:65:46: warning: Unresolved var: dashboard/routes
src/metabase/api/routes.clj:66:46: warning: Unresolved var: database/routes
src/metabase/api/routes.clj:67:46: warning: Unresolved var: dataset/routes
src/metabase/api/routes.clj:68:46: warning: Unresolved var: email/routes
src/metabase/api/routes.clj:69:65: warning: Unresolved var: embed/routes
src/metabase/api/routes.clj:70:46: warning: Unresolved var: field/routes
src/metabase/api/routes.clj:71:39: warning: Unresolved var: geojson/routes
src/metabase/api/routes.clj:72:46: warning: Unresolved var: ldap/routes
src/metabase/api/routes.clj:73:46: warning: Unresolved var: login-history/routes
src/metabase/api/routes.clj:74:46: warning: Unresolved var: premium-features/routes
src/metabase/api/routes.clj:75:46: warning: Unresolved var: metric/routes
src/metabase/api/routes.clj:76:46: warning: Unresolved var: native-query-snippet/routes
src/metabase/api/routes.clj:77:48: warning: Unresolved var: notify/routes
src/metabase/api/routes.clj:78:46: warning: Unresolved var: permissions/routes
src/metabase/api/routes.clj:79:46: warning: Unresolved var: preview-embed/routes
src/metabase/api/routes.clj:80:60: warning: Unresolved var: public/routes
src/metabase/api/routes.clj:81:46: warning: Unresolved var: pulse/routes
src/metabase/api/routes.clj:82:46: warning: Unresolved var: revision/routes
src/metabase/api/routes.clj:83:46: warning: Unresolved var: search/routes
src/metabase/api/routes.clj:84:46: warning: Unresolved var: segment/routes
src/metabase/api/routes.clj:85:39: warning: Unresolved var: session/routes
src/metabase/api/routes.clj:86:46: warning: Unresolved var: setting/routes
src/metabase/api/routes.clj:87:39: warning: Unresolved var: setup/routes
src/metabase/api/routes.clj:88:46: warning: Unresolved var: slack/routes
src/metabase/api/routes.clj:89:46: warning: Unresolved var: table/routes
src/metabase/api/routes.clj:90:46: warning: Unresolved var: task/routes
src/metabase/api/routes.clj:93:41: warning: Unresolved var: testing/routes
src/metabase/api/routes.clj:95:46: warning: Unresolved var: tiles/routes
src/metabase/api/routes.clj:96:46: warning: Unresolved var: transform/routes
src/metabase/api/routes.clj:97:46: warning: Unresolved var: user/routes
src/metabase/api/routes.clj:98:39: warning: Unresolved var: util/routes
src/metabase/api/search.clj:168:4: warning: unused binding model
src/metabase/api/search.clj:344:46: warning: unused binding k
src/metabase/api/segment.clj:69:32: warning: unused binding archived
src/metabase/api/setup.clj:61:59: warning: unused binding first_name
src/metabase/api/setup.clj:61:70: warning: unused binding last_name
src/metabase/api/setup.clj:84:29: warning: unused binding request
src/metabase/api/table.clj:199:30: warning: unused binding k
src/metabase/api/tiles.clj:112:24: warning: unused binding e
src/metabase/api/user.clj:85:41: warning: Unresolved namespace clojure.string. Are you missing a require?
src/metabase/query_processor/middleware/parameters.clj:22:25: warning: unused binding source-query
linting took 4993ms, errors: 0, warnings: 151
```
</details>

<details>
<summary>List of deprecation warnings for completeness</summary>

```
❯ clj-kondo --lint src:shared/src | grep deprecated
src/metabase/api/database.clj:80:7: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/api/database.clj:119:62: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/api/table.clj:213:15: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/automagic_dashboards/comparison.clj:39:9: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:95:50: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:211:20: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:424:54: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:593:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/automagic_dashboards/core.clj:599:30: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:1040:18: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:1117:27: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/core.clj:1127:32: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/populate.clj:89:42: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/automagic_dashboards/rules.clj:125:75: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/db/metadata_queries.clj:115:55: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver.clj:417:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver.clj:419:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver.clj:420:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver.clj:446:59: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver.clj:598:12: warning: #'metabase.driver/current-db-time is deprecated since 0.34.0
src/metabase/driver/common.clj:293:5: warning: #'metabase.driver.common/ThreadSafeSimpleDateFormat is deprecated
src/metabase/driver/common.clj:335:27: warning: #'metabase.driver.common/current-db-time-native-query is deprecated
src/metabase/driver/common.clj:336:27: warning: #'metabase.driver.common/current-db-time-date-formatters is deprecated
src/metabase/driver/common.clj:360:11: warning: #'metabase.driver.common/first-successful-parse is deprecated
src/metabase/driver/h2.clj:33:14: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/h2.clj:124:12: warning: #'metabase.driver.common/current-db-time-date-formatters is deprecated
src/metabase/driver/h2.clj:126:3: warning: #'metabase.driver.common/create-db-time-formatters is deprecated
src/metabase/driver/h2.clj:128:12: warning: #'metabase.driver.common/current-db-time-native-query is deprecated
src/metabase/driver/h2.clj:132:12: warning: #'metabase.driver/current-db-time is deprecated since 0.34.0
src/metabase/driver/h2.clj:134:10: warning: #'metabase.driver.common/current-db-time is deprecated
src/metabase/driver/mysql.clj:33:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/mysql.clj:34:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/mysql.clj:76:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/mysql.clj:165:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/mysql.clj:371:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/set-timezone-sql is deprecated since 0.35.0
src/metabase/driver/postgres.clj:73:12: warning: #'metabase.driver.common/current-db-time-date-formatters is deprecated
src/metabase/driver/postgres.clj:75:3: warning: #'metabase.driver.common/create-db-time-formatters is deprecated
src/metabase/driver/postgres.clj:77:12: warning: #'metabase.driver.common/current-db-time-native-query is deprecated
src/metabase/driver/postgres.clj:81:12: warning: #'metabase.driver/current-db-time is deprecated since 0.34.0
src/metabase/driver/postgres.clj:83:10: warning: #'metabase.driver.common/current-db-time is deprecated
src/metabase/driver/postgres.clj:420:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/set-timezone-sql is deprecated since 0.35.0
src/metabase/driver/sql.clj:28:14: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/sql.clj:34:14: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/sql.clj:36:5: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/sql/query_processor.clj:947:3: warning: #'metabase.driver.sql.query-processor.deprecated/*field-options* is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:948:3: warning: #'metabase.driver.sql.query-processor.deprecated/*source-query* is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:949:3: warning: #'metabase.driver.sql.query-processor.deprecated/*table-alias* is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:950:3: warning: #'metabase.driver.sql.query-processor.deprecated/escape-alias is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:951:3: warning: #'metabase.driver.sql.query-processor.deprecated/field->alias is deprecated since 0.41.0
src/metabase/driver/sql/query_processor.clj:952:3: warning: #'metabase.driver.sql.query-processor.deprecated/field->identifier is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:953:3: warning: #'metabase.driver.sql.query-processor.deprecated/prefix-field-alias is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:961:12: warning: #'metabase.driver.sql.query-processor.deprecated/field->identifier is deprecated since 0.42.0
src/metabase/driver/sql/query_processor.clj:963:47: warning: #'metabase.driver.sql.query-processor.deprecated/field->identifier is deprecated since 0.42.0
src/metabase/driver/sql/query_processor/deprecated.clj:55:12: warning: #'metabase.driver.sql.query-processor.deprecated/field->alias is deprecated since 0.41.0
src/metabase/driver/sql/query_processor/deprecated.clj:103:12: warning: #'metabase.driver.sql.query-processor.deprecated/escape-alias is deprecated since 0.42.0
src/metabase/driver/sql/query_processor/deprecated.clj:111:28: warning: #'metabase.driver.sql.query-processor.deprecated/escape-alias is deprecated since 0.42.0
src/metabase/driver/sql/query_processor/deprecated.clj:112:28: warning: #'metabase.driver.sql.query-processor.deprecated/escape-alias is deprecated since 0.42.0
src/metabase/driver/sql/query_processor/deprecated.clj:114:3: warning: #'metabase.driver.sql.query-processor.deprecated/escape-alias is deprecated since 0.42.0
src/metabase/driver/sql/query_processor/deprecated.clj:127:12: warning: #'metabase.driver.sql.query-processor.deprecated/prefix-field-alias is deprecated since 0.42.0
src/metabase/driver/sql_jdbc.clj:42:12: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/driver/sql_jdbc.clj:44:17: warning: #'metabase.driver.sql-jdbc.execute.old-impl/set-timezone-sql is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute.clj:152:30: warning: #'metabase.driver.sql-jdbc.execute.old-impl/set-timezone-sql is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute.clj:193:7: warning: #'metabase.driver.sql-jdbc.execute/set-time-zone-if-supported! is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute.clj:435:30: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute.clj:437:48: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute.clj:517:3: warning: #'metabase.driver.sql-jdbc.execute.old-impl/set-timezone-sql is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute.clj:518:3: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:24:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/set-timezone-sql is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:43:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:50:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:54:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:58:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:62:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/execute/old_impl.clj:66:12: warning: #'metabase.driver.sql-jdbc.execute.old-impl/read-column is deprecated since 0.35.0
src/metabase/driver/sql_jdbc/sync.clj:20:3: warning: #'metabase.driver.sql-jdbc.sync.interface/syncable-schemas is deprecated since 0.43.0
src/metabase/driver/sql_jdbc/sync/interface.clj:60:12: warning: #'metabase.driver.sql-jdbc.sync.interface/syncable-schemas is deprecated since 0.43.0
src/metabase/driver/util.clj:64:9: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/models/collection.clj:868:19: warning: #'metabase.public-settings.premium-features/enable-enhancements? is deprecated
src/metabase/models/collection/root.clj:23:17: warning: #'metabase.public-settings.premium-features/enable-enhancements? is deprecated
src/metabase/query_processor/middleware/add_implicit_joins.clj:184:17: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/query_processor/middleware/parameters/native.clj:34:11: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/query_processor/middleware/resolve_joins.clj:159:5: warning: #'metabase.query-processor.middleware.resolve-joins/maybe-resolve-source-table is deprecated
src/metabase/query_processor/middleware/results_metadata.clj:30:16: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/query_processor/timezone.clj:44:10: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/related.clj:29:32: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/related.clj:38:35: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/related.clj:100:33: warning: #'metabase.query-processor.util/normalize-token is deprecated
src/metabase/sync/fetch_metadata.clj:24:11: warning: #'metabase.driver/supports? is deprecated since 0.41.0
src/metabase/sync/sync_metadata/sync_timezone.clj:20:29: warning: #'metabase.driver/current-db-time is deprecated since 0.34.0
```

</details>

vs master

```
clj-kondo --lint src:shared/src | grep -v deprecated | wc -l
     286
```

I'm omitting deprecations since those will require a bit more work or manually excluding them with `#_:clj-kondo/ignore` in the cases where we are calling the deprecated method for legacy compatibility reasons.

Things removed in this PR:
- try without catch
- redundant `do`s
- unused bindings
- unused private vars. Following a convention of sometimes leaving the name with an underscore prepended to retain context where appropriate. There are no bright-line rules for this so just "season to taste".

This does require a slight change in how we code. Presumably everyone should have some editor integration with clj-kondo. Clojure-lsp is quite nice as well.

### CODE REVIEW

File: .lsp/config.edn [hunk 1]
```diff
@@ -1,4 +1,8 @@
 {:keep-require-at-start?        true
  :show-docs-arity-on-same-line? true
  :project-specs                 [{:project-path "deps.edn"
-                                  :classpath-cmd ["clojure" "-A:dev:ee:ee-dev:drivers:drivers-dev" "-Spath"]}]}
+                                  :classpath-cmd ["clojure" "-A:dev:ee:ee-dev:drivers:drivers-dev" "-Spath"]}]
+
+ :linters {:clojure-lsp/unused-public-var {:level :warning
+                                           :exclude-when-defined-by
+                                           #{metabase.api.common/defendpoint}}}}
```
Comment (dpsutton) [inline .lsp/config.edn:8]: this seems to not work right now which is a shame

File: src/metabase/analytics/stats.clj [hunk 2]
```diff
@@ -98,7 +98,7 @@
   (cond
     (config/config-str :rds-hostname)        :elastic-beanstalk
     (config/config-str :database-url)        :heroku ;; Putting this last as 'database-url' seems least specific
-    :default                                 :unknown))
+    :else                                    :unknown))
```
Comment (dpsutton) [inline src/metabase/analytics/stats.clj:101]: this might be configurable but seems innocent even if it clashes with people's personal preferences

Comment (escherize) [reply src/metabase/analytics/stats.clj:101]: :else is in the [clojure style guide](https://guide.clojure.style/#else-keyword-in-cond) anyway

File: .lsp/config.edn (hunk 1 above)
Comment (camsaul) [reply .lsp/config.edn:8]: You could just write a custom macroexpansion for `defendpoint` that Kondo will use to lint it

File: src/metabase/analytics/stats.clj (hunk 2 above)
Comment (camsaul) [reply src/metabase/analytics/stats.clj:101]: 👍 on standardizing on `:else`

**Review by camsaul — APPROVED** [review]

File: .lsp/config.edn (hunk 1 above)
Comment (dpsutton) [reply .lsp/config.edn:8]: yeah @borkdude will be doing that soon i think

Comment (borkdude) [reply .lsp/config.edn:8]: Will do on Thursday!

**Pull request merged**